payments/views.py

```diff
@@ -174,6 +182,33 @@ def increase_balance(event_body, topup):
     profile.money = F('money') + amount_received
     profile.save()
 
+def write_invoice_to_pdf(topup, target):
+    html = render_to_string('payments/invoice_pdf.html', {'topup': topup})
+    weasyprint.HTML(string=html).write_pdf(target)
+    return target
+
+def send_email_with_invoice(topup):
+    subject = f'Guldier - invoice no. {topup.pk}'
+    message = 'Thank you for choosing our service. Attached you will find an invoice.'
+    email = EmailMessage(subject=subject, body=message, from_email=mail_sender, to=[topup.user.email])
+    out = BytesIO()
+    write_invoice_to_pdf(topup, out)
+    email.attach(filename='invoice_{}.pdf'.format(topup.pk), content=out.getvalue(), mimetype='application/pdf')
+    email.send()
+
+
+class GetInvoiceView(LoginRequiredMixin, views.View):
+    def get(self, request, *args, **kwargs):
+        if request.user.is_staff:
+            topup_pk = self.kwargs['topup_pk']
+            topup = get_object_or_404(TopUp, pk=topup_pk)
+            response = HttpResponse(content_type='application/pdf')
```

Comment:
https://docs.djangoproject.com/en/4.0/ref/request-response/#telling-the-browser-to-treat-the-response-as-a-file-attachment